audit-github-repo-settings.py

```diff
@@ -435,9 +449,88 @@ def github_issue_exists(title):
                         # Try to avoid API rate limit
                         time.sleep(15)
                     except Exception as e:
-                        logging.error("Failed to create GitHub issue\n" + e)
+                        logging.error("Failed to create GitHub issue\n" + str(e))
                         exit(1)
 
+def check_branch_protection_graphql(owner, repo_name, token):
+    """
+    Uses GitHub's GraphQL API to fetch existing branch protection patterns (e.g. main, develop, hotfix/*).
+    Returns a list of patterns, or None on error.
+    """
+    url = "https://api.github.com/graphql"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Content-Type": "application/json"
+    }
+    query = """
+    query($owner: String!, $repo: String!) {
+      repository(owner: $owner, name: $repo) {
+        branchProtectionRules(first: 100) {
+          nodes {
+            pattern
+          }
+        }
+      }
+    }
+    """
+    variables = {
+        "owner": owner,
+        "repo": repo_name
+    }
+    try:
+        response = requests.post(url, json={'query': query, 'variables': variables}, headers=headers)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logging.error(f"GraphQL query failed for {owner}/{repo_name}: {str(e)}")
+        return None
+
+    try:
+        data = response.json()
+    except:
+        logging.error(f"Invalid JSON response for {owner}/{repo_name}")
+        return None
+
+    if ('data' not in data or
```

Comment:
What is this mean to check ?

Reply:
we are checking the GraphQL response data to confirm that the repository and branchProtectionRules fields exist and are not None. If they’re missing we log an error and return.